carrottransform/cli/subcommands/run.py

```diff
@@ -266,6 +247,37 @@ def mapstream(rules_file, output_dir, write_mode,
     nowtime = time.time()
     print("Elapsed time = {0:.5f} secs".format(nowtime - starttime))
 
+def increment_key_counts(srcfilename, metrics, tgtfile, datacol, outrecord):
```

Comment:
I'd say the code quality of the codebase is going to be hard for someone to pick up currently, so to help improve that - when adding new functions, making them typed is going to really help.